samples/apply-blend-renderer-to-hillshade/src/main/java/com/esri/arcgismaps/sample/applyblendrenderertohillshade/components/ApplyBlendRendererToHillshadeViewModel.kt

```diff
@@ -0,0 +1,231 @@
+                    outputMinValues = listOf(9.0),
+                    outputMaxValues = listOf(255.0),
+                    sourceMinValues = emptyList(),
+                    sourceMaxValues = emptyList(),
+                    noDataValues = emptyList(),
+                    gammas = emptyList(),
+                    colorRamp = colorRamp,
+                    altitude = altitude,
+                    azimuth = azimuth,
+                    slopeType = slopeType ?: SlopeType.None
+                )
+                elevationRasterLayer.renderer = renderer
+                arcGISMap.setBasemap(elevationBasemap)
+            } else {
+                // Since no ColorRamp is selected, apply BlendRenderer to imagery raster layer.
+                val renderer = BlendRenderer(
+                    elevationRaster = elevationRaster,
+                    outputMinValues = listOf(9.0),
+                    outputMaxValues = listOf(255.0),
+                    sourceMinValues = emptyList(),
+                    sourceMaxValues = emptyList(),
+                    noDataValues = emptyList(),
+                    gammas = emptyList(),
+                    colorRamp = null,
+                    altitude = altitude,
+                    azimuth = azimuth,
+                    slopeType = slopeType ?: SlopeType.None
+                )
+
+                imageryRasterLayer.renderer = renderer
+                arcGISMap.setBasemap(imageryBasemap)
+            }
+        } catch (ex: Throwable) {
+            messageDialogVM.showMessageDialog(
+                ex
+            )
+        }
+    }
+
+    /**
+     * Update the altitude value and re-apply the renderer.
+     */
+    fun updateAltitude(newAltitude: Double) {
+        altitude = newAltitude
+        updateRenderer()
+    }
+
+    /**
+     * Update the azimuth value and re-apply the renderer.
+     */
+    fun updateAzimuth(newAzimuth: Double) {
+        azimuth = newAzimuth
+        updateRenderer()
+    }
+
+    /**
+     * Update the slope type for the hillshade renderer and re-apply.
+     */
+    fun updateSlopeType(newSlopeType: SlopeType?) {
+        slopeType = newSlopeType ?: SlopeType.None
+        updateRenderer()
+    }
+
+    /**
+     * Update the selected color ramp preset index and re-apply the renderer.
+     */
+    fun updateColorRampPresetIndex(index: Int) {
+        selectedColorRampPresetIndex = index.coerceIn(0, colorRampPresets.lastIndex)
+        updateRenderer()
+    }
+
+    /**
+     * Helper that creates a Kotlin ColorRamp from the selected preset index.
+     * If the preset is "None" this returns null.
+     */
+    private fun createColorRampFromPresetIndex(index: Int): ColorRamp? {
+        return when (index) {
+            1 -> ColorRamp.create(PresetColorRampType.DemLight, 256)
+            2 -> ColorRamp.create(PresetColorRampType.DemScreen, 256)
+            3 -> ColorRamp.create(PresetColorRampType.Elevation, 256)
```

Comment:
Nitpick: I would recommend to use named args here for web doc clarity: 
```suggestion
            1 -> ColorRamp.create(type = PresetColorRampType.DemLight, size = 256)
            2 -> ColorRamp.create(type = PresetColorRampType.DemScreen, size = 256)
            3 -> ColorRamp.create(type = PresetColorRampType.Elevation, size = 256)
```